### PR TITLE
Fix null-message bug

### PR DESCRIPTION
**Pull request by ChristianGreiner**
#### Pull Request Checklist

This small pull request should fix the "Check if the log entry is null" - Issue https://github.com/AdamsLair/duality/issues/378

### CODE REVIEW

File: Duality/Utility/Log/Log.cs [hunk 1]
```diff
@@ -169,9 +169,12 @@ private void Write(LogMessageType type, string msg, object context)
...
 			Profile.TimeLog.BeginMeasure();
 
-			// Check whether the message contains null characters. If it does, crop it, because it's probably broken.
-			int nullCharIndex = msg.IndexOf('\0');
-			if (nullCharIndex != -1)
+                        if(msg == null) msg = "[null message]";
+
+                        // Check whether the message contains null characters. If it does, crop it, because it's probably broken.
+                        int nullCharIndex = msg.IndexOf('\0');
+
```
Comment (ilexp) [inline Duality/Utility/Log/Log.cs]: Code Style: Remove additional newline in line 176.

File: Duality/Utility/Log/Log.cs [hunk 2]
```diff
@@ -169,9 +169,12 @@ private void Write(LogMessageType type, string msg, object context)
 		{
 			Profile.TimeLog.BeginMeasure();
 
-			// Check whether the message contains null characters. If it does, crop it, because it's probably broken.
-			int nullCharIndex = msg.IndexOf('\0');
-			if (nullCharIndex != -1)
+                        if(msg == null) msg = "[null message]";
```
Comment (ilexp) [inline Duality/Utility/Log/Log.cs]: Code Style: Add space between `if` and opening bracket.

**Comment by ilexp** [comment]
Thanks for the PR! Added two change requests for minor code style issues.

**Comment by ilexp** [comment]
Ah, just spotted another thing: Please use tabs for indentation, instead of spaces.

**Comment by ChristianGreiner** [comment]
@ilexp 
Jeah the line spacing sucks a little bit.
I hat to indent the code two times to looks like you line indent.

My code looks like this:
```C#
Profile.TimeLog.BeginMeasure();

                        if(msg == null) msg = "[null message]";

                        // Check whether the message contains null characters. If it does, crop it, because it's probably broken.
                        int nullCharIndex = msg.IndexOf('\0');
            
                        if (nullCharIndex != -1)
			{
```

It's kinda weird...

**Comment by ilexp** [comment]
> Jeah I used tabs for indentation.

In the commit above, they're definitely spaces: In the GitHub code view, you can select them one by one, as opposed to tabs in unchanged lines, where you select per-tab. My guess would be that you pressed the Tab key, but Visual Studio is configured to use spaces, to it inserted four spaces instead. 

In Visual Studio, open `Tools / Options`, then select `Text Editor > C# > Tabs` on the right, and change the Tabs setting to `Keep Tabs`. While you're at it, you can also close the options dialog and then select `Edit / Advanced` and switch on `View White Space`. Can be quite handy when you're working in multiple projects with different tab / space preferences! :)

**Comment by ChristianGreiner** [comment]
All right. Thank you for your tips :)

**Comment by ilexp** [comment]
Let me know if you want to introduce the three requested changes in another commit to your PR branch. Otherwise, I'll close this and fix the issue on the master branch directly when I get around to it :)

**Comment by ChristianGreiner** [comment]
Omg... I don't get it.... Give me a second...

**Comment by ilexp** [comment]
I'm not sure what you did, but there's still tons of spaces, now mixed with tabs. :smile: 

Anyway, I think I'm going to just edit that line locally to give you time to sort out the spacing issue. In any case, you did get a quick preview of the overall contribution routine, so there's that :)

**Comment by ChristianGreiner** [comment]
I dont know whats ging on. In my local repo the indent is correct. As soon as I committed my code, it gets messed up. (See on github)
In the last commit it happens again so I edit my commit on github and fixed it online. On github the indent looks now correctly.

I've also set the configuration in VS exactly like you... 

Maybe I'm to dumb to commit this ting.. I dont hope so :see_no_evil:

**Comment by ilexp** [comment]
> On github the indent looks now correctly.

It looks correct at first glance, but in the latest commit, lines 174 to 176 have both tabs and spaces in their indentation. You can see it when you select the entire whitespace part in front of the line character by character. That those lines show up as changed at all is also a sign, since they didn't actually change content-wise when comparing them side-by-side.

> Maybe I'm to dumb to commit this ting.. I dont hope so :see_no_evil:

Nah, it's probably just some config messed up. If you're up for it, we can repeat this with a new branch (keeping the history clean) and new PR when you find out.

**Pull request closed without merging**